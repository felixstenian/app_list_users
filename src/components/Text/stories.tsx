import { Meta, Story } from '@storybook/react'
import Text from '.'

export default {
  title: 'Text',
  component: Text,
  args: {}
} as Meta

export const Default: Story = (args) => <Text {...args} />
