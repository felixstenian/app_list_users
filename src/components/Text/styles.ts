import styled, {
  DefaultTheme,
  FlattenSimpleInterpolation,
  css
} from 'styled-components'

import { layout, position, space, system, typography } from 'styled-system'
import { TextProps } from '.'

interface WrapperModifiers {
  [key: string]: (theme: DefaultTheme) => FlattenSimpleInterpolation
}

const wrapperModifiers: WrapperModifiers = {
  biggerTitle: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
  `,
  mediumTitle: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  smallTitle: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
  `,
  subtitle: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  paragraph: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  caption: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `
}

export const Wrapper = styled.p<TextProps>`
  ${({ theme, variant, color }) => css`
    color: ${color};
    font-family: ${theme.font.family};
    ${!!variant && wrapperModifiers[variant](theme)};
  `}
  ${space};
  ${position};
  ${typography};
  ${layout};
  ${system({ cursor: true, wordBreak: true, wordWrap: true })};
`
