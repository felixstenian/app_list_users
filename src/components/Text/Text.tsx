import * as Style from './styles'

const Text = () => (
  <Style.Wrapper>
    <h1>Text</h1>
  </Style.Wrapper>
)

export default Text
