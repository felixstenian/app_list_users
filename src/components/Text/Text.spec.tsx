import { render, screen } from '@testing-library/react'

import Text from '.'

describe('<Text />', () => {
  it('1. should render the reading', () => {
    const { container } = render(<Text />)

    expect(screen.getByRole('heading', { name: /Text/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
