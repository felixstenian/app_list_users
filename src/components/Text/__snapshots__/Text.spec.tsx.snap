// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> 1. should render the reading 1`] = `
<div
  class=""
>
  <h1>
    Text
  </h1>
</div>
`;
